seq/seq.c

```diff
@@ -40,26 +70,214 @@ int main(int argc, char *argv[]){
 		usage();
 	}
 
-	// If there's no separator set, defaults to a line break (\n)
-	separator = ( strncmp(separator, "\0", sizeof("\0")) == 0 ) ? "\n" : separator;
-
-	// If argc equals 1, stop will be defined as the first command line argument
-	// and start and step will be defaulted to 1.
-	// If argc is 2, stop will be defined as the second command line argument and
-	// start as the first.
-	// If argc is 3, stop will be defined as the third, start as the first and
-	// step as the second.
-	stop = ( argc == 1 ) ? atoi(argv[0]) : ( argc == 3 ) ? atoi(argv[2]) : atoi(argv[1]);
-	start = ( argc == 1 ) ? 1 : atoi(argv[0]);
-	step = ( argc < 3 ) ? 1 : atoi(argv[1]);
-
-	for ( count = start; count <= stop; count += step ) {
-		separator = (count == stop) ? "\n" : separator;
-		printf("%d%s", count, separator);
+	/* 
+	 * If argc equals 1, stop will be defined as the first command line argument
+	 * and start and step will be defaulted to 1.
+	 * If argc is 2, stop will be defined as the second command line argument and
+	 * start as the first.
+	 * If argc is 3, stop will be defined as the third, start as the first and
+	 * step as the second.
+	 */
+	stop = ( argc == 1 )
+		? atof(argv[0])
+		: ( argc == 3 )
+		? atof(argv[2])
+		: atof(argv[1]);
+	start = ( argc == 1 )
+		? 1
+		: atof(argv[0]);
+	step = ( argc < 3 )
+		? 1
+		: atof(argv[1]);
+	
+	if (step == 0) {
```

Comment:
I've read about it, but the most sources told to do not stress this out a lot.

Reply:
Then why stressing about zero?

Reply:
Because a zero-step sum would lead to an obvious infinite loop.

Anyway, how could this be fixed?

Reply:
I believe NaN might also cause some problems. Using a decimal integer instead of a float might be more resilient

Reply:
All the ``seq`` implementations since UNIX 8th use float as the type for counting, that would be like proposing that oil leakage in a car could be solved per not using any oil.

Reply:
It's not like proposing not using any oil, it's like proposing using real oil instead of the shitty Sonic F1 Oil

Reply:
Well, could we still have decimal numbers without having such problem with float?
I saw a solution which its centrepiece was basically to replace 0 with a "epsilon" value, small enough to be considered zero.